Zend/zend_API.c

```diff
@@ -1586,6 +1586,28 @@ ZEND_API void object_properties_init_ex(zend_object *object, HashTable *properti
 }
 /* }}} */
 
+ZEND_API zend_never_inline void zend_forbidden_dynamic_property(zend_class_entry *ce, zend_string *member) {
+	zend_throw_error(NULL, "Cannot create dynamic property %s::$%s",
+		ZSTR_VAL(ce->name), ZSTR_VAL(member));
+}
+
+ZEND_API zend_never_inline bool zend_deprecated_dynamic_property(zend_object *obj, zend_string *member) {
+	GC_ADDREF(obj);
+	zend_error(E_DEPRECATED, "Creation of dynamic property %s::$%s is deprecated",
```

Comment:
That's necessary in the version used by `zend_object_handlers` when userland is writing to a property (e.g. opcode handler), but not for unserialization.

(The object isn't set globally - calls to both `__wakeup` and `unserialize` are deferred to the end of unserialize() to avoid this)

The original check is meant to handle:

- There's an object accessible from the global scope
- Userland attempts to write a deprecated dynamic property to that object
- That triggers a userland error handler
- The error handler releases all the references to that object (e.g. `unset($GLOBALS['globalVar']);`, which would have freed the object
- The property writer resumes and (if not for the GC_DELREF check below) tries to write to a freed object